src/services/epg/viewNexaEpg.service.ts

```diff
@@ -0,0 +1,79 @@
+import { object, string } from 'yup';
+import { parse } from 'date-fns';
+import { injectable } from 'inversify';
+
+import EpgProviderService from './epgProvider.service';
+
+import type { PlaylistItem } from '#types/playlist';
+import { logDev } from '#src/utils/common';
+import type { EpgProgram } from '#types/epg';
+
+const viewNexaEpgProgramSchema = object().shape({
+  'episode-num': object().shape({
+    '#text': string().required(),
+  }),
+  title: object().shape({
+    '#text': string().required(),
+  }),
+  desc: object().shape({
+    '#text': string(),
+  }),
+  icon: object().shape({
+    src: string(),
+  }),
+  start: string().required(),
+  stop: string().required(),
+});
+
+const parseData = (date: string): string => parse(date, 'yyyyMdHms xxxx', new Date()).toISOString();
+
+@injectable()
+export default class ViewNexaEpgService extends EpgProviderService {
+  /**
+   * Validate the given data with the viewNexaProgramSchema and transform it into an EpgProgram
+   */
+  transformProgram = async (data: unknown): Promise<EpgProgram> => {
+    const program = await viewNexaEpgProgramSchema.validate(data);
+
+    return {
+      id: program['episode-num']['#text'],
```

Comment:
Do we need to worry about null refs here (i.e. if `program['episode-num']` is null) or does the parser handle that?

Reply:
I would expect all properties to be present in the xml response. We have a `yup` validation schema which will be triggered when some of required fields are missing.

Reply:
OK, and does the yup schema validation throw a fatal error or do we just not show the epg for broken schemas?

Reply:
The page itself is still shown. We filter items where the validation failed (`parseSchedule` method)